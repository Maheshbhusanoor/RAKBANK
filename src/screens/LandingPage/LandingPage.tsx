import React, {useCallback, useEffect, useRef, useState} from 'react';
import {Animated, Text, TouchableOpacity} from 'react-native';
import {Button} from '../../components';
import {
  Container,
  SubContainer,
  BottomContainer,
  FormContainer,
  Input,
  InfoTitle,
  InfoContainer,
  Info,
  RegisterContainer,
  Register,
} from './styles';
import {perfectSize} from '../../themes/screen';
import colors from '../../themes/colors';
import Icon from 'react-native-vector-icons/FontAwesome';
import {useDispatch} from 'react-redux';
import createMockServer from '../../services/mirage';
import {
  fetchLoginDataRequest,
  fetchUserCurrentLocationRequest,
} from '../../store/actions';
import {useOnRequestSuccess, useRequestLoading} from 'react-redux-help-kit';
import {loginApiPayloadModel} from 'src/types';
export const LandingPage = ({navigation}) => {
  const dispatch = useDispatch();
  const [userId, setUserID] = useState('');
  const [navigateTo, setNavigateTo] = useState(false);

  const [password, setPassword] = useState('');
  const subContainerMarginTop = useRef(new Animated.Value(0)).current;
  const bottomContainerOpacity = useRef(new Animated.Value(1)).current;
  const backArrowLeft = useRef(new Animated.Value(-50)).current;

  useEffect(() => {
    createMockServer();
  }, []);
  useEffect(() => {
    if (navigateTo) {
      navigation.navigate('Home');
    }
  }, [navigateTo]);
  const onLogin = () => {
    Animated.parallel([
      Animated.timing(subContainerMarginTop, {
        toValue: perfectSize(-2400),
        duration: 1000,
        useNativeDriver: false,
      }),
      Animated.timing(bottomContainerOpacity, {
        toValue: 0,
        duration: 1000,
        useNativeDriver: false,
      }),
      Animated.timing(backArrowLeft, {
        toValue: 10,
        duration: 1000,
        useNativeDriver: false,
      }),
    ]).start();
  };

  const {loading} = useRequestLoading(fetchLoginDataRequest);

  const loginWithUserNamePassword = useCallback(() => {
    setNavigateTo(false);

    dispatch(fetchUserCurrentLocationRequest());
  }, [dispatch]);

  useOnRequestSuccess(
    fetchUserCurrentLocationRequest,
    (data: loginApiPayloadModel) => {
      data.userConfigData.userId = userId;
      data.userConfigData.password = userId;
      dispatch(fetchLoginDataRequest(data));
    },
  );
  useOnRequestSuccess(fetchLoginDataRequest, () => {
    setUserID('');
    setPassword('');
    setNavigateTo(true);
  });
  const onBack = () => {
    Animated.parallel([
      Animated.timing(subContainerMarginTop, {
        toValue: 0,
        duration: 1000,
        useNativeDriver: false,
      }),
      Animated.timing(bottomContainerOpacity, {
        toValue: 1,
        duration: 1000,
        useNativeDriver: false,
      }),
      Animated.timing(backArrowLeft, {
        toValue: -50,
        duration: 1000,
        useNativeDriver: false,
      }),
    ]).start();
    setUserID('');
    setPassword('');
  };
  return (
    <Container pointerEvents={loading ? 'none' : 'auto'}>
      <Animated.View
        style={{
          marginTop: subContainerMarginTop,
        }}>
        <SubContainer>
          <InfoTitle>RAKBANK</InfoTitle>
          <InfoContainer>
            <Info>
              Everything you love about Digital Banking in a smarter, simpler
              design
            </Info>
          </InfoContainer>
        </SubContainer>
      </Animated.View>
      <Animated.View
        style={{
          left: backArrowLeft,
          position: 'absolute',
          top: perfectSize(80),
          zIndex: 1,
        }}>
        <TouchableOpacity onPress={() => onBack()}>
          <Icon name="long-arrow-left" size={28} />
        </TouchableOpacity>
      </Animated.View>

      <RegisterContainer>
        <Register>Register</Register>
      </RegisterContainer>

      <Animated.View
        style={{
          opacity: bottomContainerOpacity,
        }}>
        <BottomContainer>
          <Button
            onPress={() => onLogin()}
            title="Login with User ID"
            background={true}
          />
          <Button title="Quick balance" top={'4%'} />
        </BottomContainer>
      </Animated.View>
      <Animated.View>
        <FormContainer>
          <Input
            placeholder="Enter user ID"
            onChangeText={userId => setUserID(userId)}
            value={userId}
            placeholderTextColor={colors.placeholderColor}
          />
          <Input
            placeholder="Enter Password"
            style={{
              marginTop: perfectSize(20),
            }}
            onChangeText={password => setPassword(password)}
            value={password}
            placeholderTextColor={colors.placeholderColor}
            secureTextEntry={true}
          />
          <Button
            background={true}
            title={'Submit'}
            onPress={() => loginWithUserNamePassword()}
            disabled={!userId || !password}
            top={'4%'}
            loading={loading}
          />
        </FormContainer>
      </Animated.View>
    </Container>
  );
};
