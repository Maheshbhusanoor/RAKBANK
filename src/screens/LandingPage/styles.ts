import styled from 'styled-components/native';
import {perfectSize} from '../../themes/screen';
export const Container = styled.View`
  flex: 1;
  height: 100%;
  width: 100%;
  padding: 10%;
`;
interface Props {
  background?: boolean;
  top?: string;
}
export const TopContainer = styled.Text<Props>`
  height: 40%;
  width: 100%;
  align-items: center;
  justify-content: center;
`;

export const SubContainer = styled.View`
    height: 75%
    width: 100%;
    background-color: transparent;
`;
export const InfoContainer = styled.View`
  width: 50%;
`;
export const Info = styled.Text`
  fon-size: ${perfectSize(18)};
  margin-top: 2%;
`;

export const InfoTitle = styled.Text`
  font-size: ${perfectSize(25)};
  margin-top: 45%;
  font-weight: bold;
`;

export const BottomContainer = styled.View`
  height: 40%;
  width: 100%;
  align-items: center;
  justify-content: center;
`;

export const FormContainer = styled.View`
  /* flex: 1; */
  height: 50%;
  width: 100%;
  align-items: center;
  justify-content: center;
`;

export const Input = styled.TextInput`
  height: ${perfectSize(60)};
  border-radius: ${perfectSize(7)};
  width: 100%;
  background-color: white;
  font-size: ${perfectSize(19)};
  padding-left: 3%;
`;
export const RegisterContainer = styled.View`
  /* padding: 6%; */
  border-width: 1;
  height: 6%;
  width: 30%;
  align-items: center;
  justify-content: center;
  border-radius: 4;
  position: absolute;
  top: ${perfectSize(80)};
  right: 10;
`;
export const Register = styled.Text`
  font-size: 18;
`;
