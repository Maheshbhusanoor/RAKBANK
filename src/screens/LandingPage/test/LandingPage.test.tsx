import {LandingPage} from '../LandingPage';
import { render,act} from 'react-native-testing-library'
import { create } from 'react-test-renderer';
import { shallow } from "enzyme";

beforeEach(jest.clearAllMocks)

describe('Landing Page', () => {
    const wrapper = shallow(<LandingPage/>);

    it('Render Landing Page', async () => {
          const tree = render(<LandingPage />)

          expect(tree).toMatchSnapshot()
    });
    it('Create Landing Page', () => {
        let root
        act(() => {
            root = create(<LandingPage />) 
        })
         expect(root).toMatchSnapshot()
    })

   it('Login Button Props Validation for ', function () {
        expect(wrapper.find('Button').first().props()).toEqual({
      onPress: expect.any(Function),
      background:true,
      title:"Login with User ID"
    });
    });
 
    it('Submit Button Props Validation ', function () {
        expect(wrapper.find('Button').last().props()).toEqual({
        onPress: expect.any(Function),
        background:true,
        title:"Submit",
        disabled: true,
        loading:undefined,
        top:'4%'
        });
    });
    it('Buttons title to be defined', function () {
        wrapper.find('Button').forEach( (node) => {
        expect(node.prop('title')).toBeDefined()
        })
    });
 
});
