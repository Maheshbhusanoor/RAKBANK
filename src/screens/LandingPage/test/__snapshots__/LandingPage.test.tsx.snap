// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Create Landing Page 1`] = `
<View
  pointerEvents="auto"
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "paddingBottom": "10%",
        "paddingLeft": "10%",
        "paddingRight": "10%",
        "paddingTop": "10%",
        "width": "100%",
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "marginTop": 0,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": "75%",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 38.846579013227505,
              "fontWeight": "bold",
              "marginTop": "45%",
            },
          ]
        }
      >
        RAKBANK
      </Text>
      <View
        style={
          Array [
            Object {
              "width": "50%",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fonSize": 27.969536889523805,
                "marginTop": "2%",
              },
            ]
          }
        >
          Everything you love about Digital Banking in a smarter, simpler design
        </Text>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      Object {
        "left": -50,
        "position": "absolute",
        "top": 124.30905284232801,
        "zIndex": 1,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 28,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï…·
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": "6%",
          "justifyContent": "center",
          "position": "absolute",
          "right": 10,
          "top": 124.30905284232801,
          "width": "30%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 18,
          },
        ]
      }
    >
      Register
    </Text>
  </View>
  <View
    collapsable={false}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": "40%",
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 30,
            "display": "flex",
            "height": 40,
            "justifyContent": "center",
            "marginTop": 0,
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          background={true}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
            ]
          }
        >
          Login with User ID
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 30,
            "display": "flex",
            "height": 40,
            "justifyContent": "center",
            "marginTop": "4%",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 20,
              },
            ]
          }
        >
          Quick balance
        </Text>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={Object {}}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": "50%",
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Enter user ID"
        placeholderTextColor="rgba(66,76,89,1)"
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 10.877042123703701,
              "fontSize": 29.523400050052903,
              "height": 93.23178963174601,
              "paddingLeft": "3%",
              "width": "100%",
            },
          ]
        }
        value=""
      />
      <TextInput
        onChangeText={[Function]}
        placeholder="Enter Password"
        placeholderTextColor="rgba(66,76,89,1)"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 10.877042123703701,
              "fontSize": 29.523400050052903,
              "height": 93.23178963174601,
              "paddingLeft": "3%",
              "width": "100%",
            },
            Object {
              "marginTop": 31.077263210582004,
            },
          ]
        }
        value=""
      />
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 30,
            "display": "flex",
            "height": 40,
            "justifyContent": "center",
            "marginTop": "4%",
            "opacity": 0.3,
            "width": "100%",
          }
        }
      >
        <Text
          background={true}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
            ]
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Landing Page Render Landing Page 1`] = `
<View
  pointerEvents="auto"
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "paddingBottom": "10%",
        "paddingLeft": "10%",
        "paddingRight": "10%",
        "paddingTop": "10%",
        "width": "100%",
      },
    ]
  }
>
  <View
    collapsable={false}
    style={
      Object {
        "marginTop": 0,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": "75%",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 38.846579013227505,
              "fontWeight": "bold",
              "marginTop": "45%",
            },
          ]
        }
      >
        RAKBANK
      </Text>
      <View
        style={
          Array [
            Object {
              "width": "50%",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fonSize": 27.969536889523805,
                "marginTop": "2%",
              },
            ]
          }
        >
          Everything you love about Digital Banking in a smarter, simpler design
        </Text>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      Object {
        "left": -50,
        "position": "absolute",
        "top": 124.30905284232801,
        "zIndex": 1,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 28,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï…·
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": "6%",
          "justifyContent": "center",
          "position": "absolute",
          "right": 10,
          "top": 124.30905284232801,
          "width": "30%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 18,
          },
        ]
      }
    >
      Register
    </Text>
  </View>
  <View
    collapsable={false}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": "40%",
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 30,
            "display": "flex",
            "height": 40,
            "justifyContent": "center",
            "marginTop": 0,
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          background={true}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
            ]
          }
        >
          Login with User ID
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 30,
            "display": "flex",
            "height": 40,
            "justifyContent": "center",
            "marginTop": "4%",
            "opacity": 1,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 20,
              },
            ]
          }
        >
          Quick balance
        </Text>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={Object {}}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": "50%",
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Enter user ID"
        placeholderTextColor="rgba(66,76,89,1)"
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 10.877042123703701,
              "fontSize": 29.523400050052903,
              "height": 93.23178963174601,
              "paddingLeft": "3%",
              "width": "100%",
            },
          ]
        }
        value=""
      />
      <TextInput
        onChangeText={[Function]}
        placeholder="Enter Password"
        placeholderTextColor="rgba(66,76,89,1)"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 10.877042123703701,
              "fontSize": 29.523400050052903,
              "height": 93.23178963174601,
              "paddingLeft": "3%",
              "width": "100%",
            },
            Object {
              "marginTop": 31.077263210582004,
            },
          ]
        }
        value=""
      />
      <View
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "black",
            "borderRadius": 30,
            "display": "flex",
            "height": 40,
            "justifyContent": "center",
            "marginTop": "4%",
            "opacity": 0.3,
            "width": "100%",
          }
        }
      >
        <Text
          background={true}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
            ]
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</View>
`;
